src/pyfaf/queries.py

```diff
@@ -1416,3 +1371,123 @@ def get_all_report_hashes(db, date_from=None,
         query = (query.slice(limit_from, limit_to))
 
     return query.all()
+
+def get_user_by_mail(db, mail):
+    """
+    Return query for User objects for given mail.
+    """
+    return db.session.query(User).filter(User.mail == mail)
+
+
+def get_reportarchives_by_username(db, username):
+    """
+    Returns query for ReportArchive objects for given username.
+    """
+    return db.session.query(ReportArchive).filter(ReportArchive.username == username)
+
+def get_problemreassigns_by_username(db, username):
+    """
+    Returns query for ProblemReassign objects for given username.
+    """
+    return db.session.query(ProblemReassign).filter(ProblemReassign.username == username)
+
+def get_bugzillas_by_uid(db, user_id):
+    """
+    Return query for BzBug for given user_id.
+    """
+    return db.session.query(BzBug).filter(BzBug.creator_id == user_id)
+
+def get_bzattachments_by_uid(db, user_id):
+    """
+    Return query for BzAttachment objects for given user_id.
+    """
+    return db.session.query(BzAttachment).filter(BzAttachment.user_id == user_id)
+
+def get_bzbugccs_by_uid(db, user_id):
+    """
+    Return query for BzBugCc objects for given user_id.
+    """
+    return db.session.query(BzBugCc).filter(BzBugCc.user_id == user_id)
+
+def get_bzbughistory_by_uid(db, user_id):
+    """
+    Return query for BzBugHistory objects for given user_id.
+    """
+    return db.session.query(BzBugHistory).filter(BzBugHistory.user_id == user_id)
+
+def get_bzcomments_by_uid(db, user_id):
+    """
+    Return query for BzComment objects for given user_id.
+    """
+    return db.session.query(BzComment).filter(BzComment.user_id == user_id)
+
+def delete_bz_user(db, user_mail):
+    """
+    Delete BzUser instance if there is a user in the database with `user_mail` mail.
+    """
+    db.session.query(BzUser).filter(BzUser.email == user_mail).delete(False)
+
+def delete_bugzilla(db, bug_id):
+    """
+    Delete Bugzilla for given bug_id.
+    """
+    query = (db.session.query(BzBug)
+             .filter(BzBug.duplicate == bug_id)
+             .all())
+
+    for bgz in query:
+        bgz.duplicate = None
+
+    db.session.query(BzComment).filter(BzComment.bug_id == bug_id).delete(False)
```

Comment:
would be nice to also delete the lob (if there is one)

Reply:
Will do.